modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js

```diff
@@ -0,0 +1,280 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {DragDropContext as dragDropContext} from 'react-dnd';
+import HTML5Backend from 'react-dnd-html5-backend';
+import CriteriaSidebar from '../criteria_sidebar/CriteriaSidebar.es';
+import {sub} from '../../utils/utils.es';
+import CriteriaBuilder from './CriteriaBuilder.es';
+import {buildQueryString, translateQueryToCriteria} from '../../utils/odata.es';
+import {CONJUNCTIONS} from '../../utils/constants.es';
+import Conjunction from './Conjunction.es';
+
+class ContributorBuilder extends React.Component {
+	constructor(props) {
+		super(props);
+
+		const {initialContributors, propertyGroups} = props;
+
+		const contributors = initialContributors && initialContributors.map(
+			c => {
+				const propertyGroup = propertyGroups &&
+					propertyGroups.find(
+						propertyGroup => c.propertyKey === propertyGroup.propertyKey
+					);
+
+				return {
+					conjunctionId: c.conjunctionId || CONJUNCTIONS.AND,
+					conjunctionInputId: c.conjunctionInputId,
+					criteriaMap: c.initialQuery ?
+						translateQueryToCriteria(c.initialQuery) :
+						null,
+					inputId: c.inputId,
+					modelLabel: propertyGroup && propertyGroup.name,
+					properties: propertyGroup && propertyGroup.properties,
+					propertyKey: c.propertyKey,
+					query: c.initialQuery
+				};
+			}
+		);
+
+		this.state = {
+			conjunctionName: CONJUNCTIONS.AND,
+			contributors,
+			editingId: undefined,
+			newPropertyKey: propertyGroups.length && propertyGroups[0].propertyKey
+		};
+	}
+
+	_onCriteriaEdit = (id, editing) => {
```

Comment:
See comment above. This should be `handleCriteriaEdit`. 

To easily catch these try adding this eslint rule, https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-handler-names.md